### PR TITLE
nix-zsh-completions package and module support 

### PR DESCRIPTION
**Pull request by spwhitt**
1. Adds a package for nix-zsh-completions
2. Configures zsh's fpath to include completions installed through Nix packages
3. Adds an enableCompletion option ZSH module

I set up the enableCompletion option to also pull in the nix-zsh-completions package, under the assumption that anyone using completions on NixOS is probably interested in completions for the Nix utilities as well. If anyone doesn't like this, feel free to let me know and I'll remove that line, and people will have to add nix-zsh-completions to their `systemPackages` manually.

@jagajaga @offlinehacker @garbas

### CODE REVIEW

File: nixos/modules/programs/zsh/zsh.nix [hunk 1]
```diff
@@ -73,6 +73,14 @@ in
         type = types.lines;
       };
 
+      enableCompletion = mkOption {
```
Comment (jagajaga) [inline nixos/modules/programs/zsh/zsh.nix]: In my opinion `True` as a default here is better :)

Comment (spwhitt) [inline nixos/modules/programs/zsh/zsh.nix]: If you say so :)

**Comment by jagajaga** [comment]
:+1:

**Comment by globin** [comment]
see #11155

**Comment by spwhitt** [comment]
@abbradar thanks - it must have worked for me because my desktop environment links `/share`. The actual required path is `/share/zsh/$ZSH_VERSION/functions` so I'm guessing I should just link all of `/share/zsh` since I won't be able to use `$ZSH_VERSION` inside `pathsToLink` - is this correct?

**Comment by abbradar** [comment]
Strange, it works for me with `/share/zsh/functions` but I guess there's both that and `$ZSH_VERSION` path so we actually need four paths to add to `pathsToLink`. For `$ZSH_VERSION` -- what do you think of using `lib.getVersion zsh` to make this path static?

**Comment by spwhitt** [comment]
My system doesn't have `/share/zsh/functions` and if yours does it's either a bug in some other package or it needs to be added to `$fpath` as well... using lib.getVersion sounds brittle, simple formatting changes could break it. `/share/zsh` doesn't have anything we don't want to link anyway (in the zsh package), so linking the whole thing shouldn't effect performance. I argume we play it safe and link `/share/zsh`

**Comment by abbradar** [comment]
`nix-zsh-completions` uses this path, for example. Also my `fpath` already contains needed paths, added by the very https://github.com/spwhitt/nixpkgs/commit/ff58711bda11e4f79fed80e209fd91cff3ff62ef#diff-0740f57e63af61694d14796286cb9204R115 ^_^.

OK, let's link the whole thing -- shouldn't hurt.

**Comment by abbradar** [comment]
Oh, sorry, it uses `/share/zsh/site-functions` -- sleepy me.

**Pull request merged**